aptos-move/framework/aptos-framework/sources/BigVector.move

```diff
@@ -0,0 +1,207 @@
+module AptosFramework::BigVector {
+    use Std::Vector;
+    use AptosFramework::Table::{Self, Table};
+
+    /// The index into the vector is out of bounds
+    const EINDEX_OUT_OF_BOUNDS: u64 = 0;
+
+    /// Index of the value in the buckets.
+    struct BigVectorIdx has copy, drop, store {
+        bucket_idx: u64,
+        vec_idx: u64,
+    }
+
+    /// A Scalable vector implementation based on tables, elements are grouped into buckets with `bucket_size`.
+    struct BigVector<T> has store {
+        buckets: Table<u64, vector<T>>,
+        next_idx: BigVectorIdx,
+        bucket_size: u64
+    }
+
+    /// Regular Vector API
+
+    /// Create an empty vector.
+    public fun new<T: store>(bucket_size: u64): BigVector<T> {
```

Comment:
Does it make sense to define the bucket size depending on the size of individual elements? If Move doesn't support something like `size_of` then maybe we can ask user to pass the estimates size of individual elements?

Reply:
user can always estimate the size and choose proper bucket size before constructing, this seems providing more flexibility?